src/lsp/server.ml

```diff
@@ -39,6 +41,44 @@ class dolmen_lsp_server =
     method on_notif_doc_did_change ~notify_back d _c ~old_content:_old ~new_content =
       self#_on_doc ~notify_back d.uri new_content
 
+    method! on_notification_unhandled
+        ~notify_back (n:Lsp.Client_notification.t) =
+      match n with
+      | Lsp.Client_notification.ChangeConfiguration { settings; } ->
+        begin try
+            let prelude_files =
+              begin match settings with
+                | `Assoc [ "preludes", `List l ] ->
+                  List.fold_left (
+                    fun acc s ->
+                      match s with
+                      | `String f -> f :: acc
+                      | _ ->
+                        failwith (
```

Comment:
You should not use `failwith`/`Failure` as a jump mechanism (it's better used as a way to abort with a message), use a different exception instead (for instance, by declaring a new one).

Reply:
Done.